units/src/amount.rs

```diff
@@ -1349,8 +1357,8 @@ impl SignedAmount {
     ///
     /// Does not include the denomination.
     #[rustfmt::skip]
-    pub fn fmt_value_in(self, f: &mut dyn fmt::Write, denom: Denomination) -> fmt::Result {
-        fmt_satoshi_in(self.unsigned_abs().to_sat(), self.is_negative(), f, denom, false, FormatOptions::default())
+    pub fn fmt_value_in(self, f: &mut dyn fmt::Write, denom: Denomination, options: FormatOptions) -> fmt::Result {
```

Comment:
In 93037b184570fd52ff111c232c92baf2da1d4475:

I don't think we should publicly expose `FormatOptions` like this. To a user it's confusing to have two different sets of options, those on `f` and those on `options`.

Reply:
My understanding of this is that it is also used when writing to a buffer like a String

Reply:
My understanding is that it is also used in writing to a buffer like a String

```rust
#[cfg(feature = "alloc")]
    pub fn to_string_in(self, denom: Denomination) -> String {
        let mut buf = String::new();
        self.fmt_value_in(&mut buf, denom, FormatOptions::default()).unwrap();
        buf
    }
```
 So adding `options: FormatOptions` seemed like the best idea at the time since once can't get `.precision()` from `fmt::Write` the same way the `impl Display for Amount` handles it:
```rust
    /// Format the given satoshi amount in the given denomination.
fn fmt_satoshi_in(
    satoshi: u64,
    negative: bool,
    f: &mut dyn fmt::Write,
    denom: Denomination,
    show_denom: bool,
    options: FormatOptions,
) -> fmt::Result {
```

Reply:
Is there a way to get the precision from an `&mut dyn fmt::Write` ?

Reply:
I have moved to using `fmt_satoshi_in()` since the outcome is the same without publicly exposing `FormatOptions `

Reply:
Ah, ok, I understand the API problem you're trying to solve.

And no, [the `Write` trait has no facility for getting formatter options](https://doc.rust-lang.org/std/fmt/trait.Write.html).

But I think this business of using `&mut dyn Write` in a public-facing API was probably a mistake to begin with ... lemme dig into this.